main.go

```diff
@@ -64,22 +64,40 @@ func main() {
 	go serveMetrics(cfg.MetricsAddress)
 	go handleSigterm(stopChan)
 
-	client, err := newClient(cfg)
-	if err != nil {
-		log.Fatal(err)
+	var fakeSrcOnly = false
+	for _, svcType := range cfg.Sources {
+		switch svcType {
+		case "fake":
+			fqdn := cfg.FqdnTemplate
+			fakeSource, err := source.NewFakeSource(fqdn)
+			if err != nil {
+				log.Fatal(err)
+			}
+			source.Register("fake", fakeSource)
+
+			fakeSrcOnly = true
+			log.Warn("Fake source registered; will not register any other sources.")
+		}
 	}
 
-	serviceSource, err := source.NewServiceSource(client, cfg.Namespace, cfg.FqdnTemplate, cfg.Compatibility)
-	if err != nil {
-		log.Fatal(err)
-	}
-	source.Register("service", serviceSource)
+	if !fakeSrcOnly {
+		client, err := newClient(cfg)
```

Comment:
👍  for making the Kubernetes client optional. However, this looks a bit scary. Let's iterate on making it a bit cleaner.